students/BrianNisonger/session4/file_excercise.py

```diff
@@ -0,0 +1,44 @@
+import pathlib
+import collections
+
+def file_1():
+    pth=pathlib.Path("./")
+    for f in pth.iterdir():
+        print(f'{pth.absolute()}\{f}')
+
+def file_2(source,dest):        
+    with open(source, 'rb') as infile, open(dest, 'wb') as outfile:
+        outfile.write(infile.read())
+
+def file_3(source):
+    name_dict={}
+    lang_list=[]
+    with open(source) as f:
+        lines=f.readlines()
+    for line in lines:
+        line_out=line.split(":")
+        if line_out[0] !="Name":
+            name_dict[line_out[0]]=line_out[1]
+    for name in name_dict:
+        if name_dict[name] !="":
+            languages=(name_dict[name].split(","))
+            for language in languages:
+                if language.islower():
+                    language=language.strip()
+                    lang_list.append(language)
+    c=collections.Counter(lang_list)
```

Comment:
nice find!

though rather than building up the list, and then passing it to counter, you could be adding stuff to the counter as you loop through the file.